packages/generate/src/edgeql-js/generateOperatorTypes.ts

```diff
@@ -314,10 +323,18 @@ export function generateOperators({
     }
   } else if (args.length === 5) {
     if (typeof args[1] === "string" && typeof args[3] === "string") {
+      // Python-style if-else
       op = \`\${args[1]}_\${args[3]}\`;
       params = [args[0], args[2], args[4]];
       defs = overloadDefs.Ternary[op];
     }
+  } else if (args.length === 6) {
+    // Functional-style if-then-else
+    if (typeof args[0] === "string" && typeof args[2] === "string" && typeof args[4] === "string") {
+      op = \`\${args[0]}_\${args[4]}\`;
+      params = [args[3], args[1], args[5]];
+      defs = overloadDefs.Ternary[op];
+    }
```

Comment:
Note that this effectively maps both forms into the same operator which gets rendered as the same old `if..else` python style form.

In the future we can thread through a `__form__` property in the expression to output different forms here based on the input form or server version or capabilities, etc.